### PR TITLE
Convert DBClient.connect() to be a context manager

### PR DESCRIPTION
**Pull request by onyb**
### What was wrong?

Ref: #1397 

### How was it fixed?

The new API for `DBClient.connect` is as follows:

```py
with DBClient.connect(...) as db:
    ...
```

#### Cute Animal Picture

put a cute animal picture link inside the parentheses

### CODE REVIEW

File: trinity/db/manager.py [hunk 1]
```diff
@@ -348,12 +348,15 @@ def close(self) -> None:
...
+    @contextlib.contextmanager
+    def connect(cls, path: pathlib.Path, timeout: int = 5) -> Iterator["DBClient"]:
         wait_for_ipc(path, timeout)
         s = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
         cls.logger.debug("Opened connection to %s: %s", path, s)
         s.connect(str(path))
-        return cls(s)
+        db = cls(s)
+        yield db
+        db.close()
```
Comment (pipermerriam) [inline trinity/db/manager.py]: These two lines should typically be written using a `try/finally` to ensure the `close` happens even in the case of an exception.

```python
try:
    yield
finally:
    db.close()
```

Comment (onyb) [reply trinity/db/manager.py]: Fixed

**Comment by onyb** [comment]
@pipermerriam Not sure if the `py{36,37}-wheel-cli` tests are expected to fail. Are the errors on the CI something you recognize?

**Comment by cburgdorf** [comment]
@onyb all CI jobs are expected to succeed. That said, the `py{36,37}-wheel-cli` wheel jobs have been unstable for some weeks. It is mostly [the pesty `EOFError` described here](https://github.com/ethereum/trinity/issues/1435) when it fails. But in your case, it is something different. It looks like the tests that spin up Trinity and then attach to it via `trinity attach` to run assertions against the `web3` console fail. I checked it locally and the error persists so it seems legit.

You can run `pytest tests/integration/test_trinity_cli.py -k 'test_web3_commands_via_attached_console'` to dig in locally.

**Pull request closed without merging**